apps/api-extractor-model/src/items/ApiItem.ts

```diff
@@ -84,6 +86,7 @@ export class ApiItem {
   /** @virtual */
   public serializeInto(jsonObject: Partial<IApiItemJson>): void {
     jsonObject.kind = this.kind;
+    jsonObject.canonicalReference = this.canonicalReference.toString();
```

Comment:
It's not possible.  This PR ensures that `ApiItem.canonicalReference` can never be `undefined` because it is based on the `ApiItem` tree itself, rather than being passed in as a parameter.